File: CS207b/L/w01/improved/model/Rental.py
class Rental:
    __movie = None
    __days_rented = None

    def __init__(self, movie, days_rented):
        self.__movie = movie
        self.__days_rented = days_rented

    def get_movie(self):
        return self.__movie

    def get_days_rented(self):
        return self.__days_rented

    def get_charge(self):
        return self.__movie.get_charge(self.__days_rented)

File: CS207b/L/w04/pkg/customer/rental.go
package customer

import (
	"207w04/movie"
)

type Rental struct {
	movie      movie.Movie
	daysRented int
}

func NewRental(movie movie.Movie, daysRented int) Rental {
	return Rental{movie: movie, daysRented: daysRented}
}

func (r *Rental) GetMovie() movie.Movie {
	return r.movie
}

func (r *Rental) GetDaysRented() int {
	return r.daysRented
}

func (r *Rental) GetCharge() float64 {
	return r.movie.GetCharge(r.daysRented)
}
